solr/core/src/java/org/apache/solr/servlet/LoadAdminUiServlet.java

```diff
@@ -24,29 +31,26 @@
 import org.apache.solr.core.CoreContainer;
 import org.apache.solr.core.SolrCore;
 
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.OutputStreamWriter;
-import java.io.Writer;
-import java.nio.charset.StandardCharsets;
-
 /**
  * A simple servlet to load the Solr Admin UI
  * 
  * @since solr 4.0
  */
 public final class LoadAdminUiServlet extends BaseSolrServlet {
 
+  // check system properties for whether or not admin UI is disabled, default is false
+  private static final boolean disabled = Boolean.parseBoolean(System.getProperty("disableAdminUI", "false"));
+
   @Override
-  public void doGet(HttpServletRequest _request,
-                    HttpServletResponse _response)
-      throws IOException {
+  public void doGet(HttpServletRequest _request, HttpServletResponse _response) throws IOException {
     HttpServletRequest request = SolrDispatchFilter.closeShield(_request, false);
     HttpServletResponse response = SolrDispatchFilter.closeShield(_response, false);
-    
+
+    if(disabled){
```

Comment:
Would this make sense to do before the close shield methods?

Reply:
Not in its current state because response object won't exist, but I could refactor if you have a good reason for that. Or, if I think of one. Gonna peep those methods again now.

Reply:
So, I got pulled away by some work. I'll update. I think it makes sense to drop the response before the close shield if possible.

Reply:
I'm testing it now.

Reply:
Handling the raw response works better. Looking at the snippet on this page originally I didn't have the full context. Good suggestion. Thanks 👍